ddl/column.go

```diff
@@ -31,15 +31,20 @@ import (
 	"github.com/pingcap/tidb/util/types"
 )
 
-func (d *ddl) adjustColumnOffset(columns []*model.ColumnInfo, indices []*model.IndexInfo, offset int, added bool) {
+func (d *ddl) adjustColumnOffset(columns []*model.ColumnInfo, indices []*model.IndexInfo, offset int, lastColOffset int, added bool) {
```

Comment:
Yes. During the period the state of new added columns become to StatePublic from StateNone,  New columns is hidden because it's offset is the end of last column.  So that new record is added to the same table in other session with the offset of old columns.